WordPress/src/test/java/org/wordpress/android/ui/posts/prepublishing/PrepublishingViewModelTest.kt

```diff
@@ -0,0 +1,137 @@
+package org.wordpress.android.ui.posts.prepublishing
+
+import com.nhaarman.mockitokotlin2.mock
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.wordpress.android.BaseUnitTest
+import org.wordpress.android.ui.posts.PrepublishingHomeItemUiState.ActionType
+import org.wordpress.android.ui.posts.PrepublishingNavigationTarget
+import org.wordpress.android.ui.posts.PrepublishingScreen.HOME
+import org.wordpress.android.ui.posts.PrepublishingScreen.PUBLISH
+import org.wordpress.android.ui.posts.PrepublishingScreen.TAGS
+import org.wordpress.android.ui.posts.PrepublishingScreen.VISIBILITY
+import org.wordpress.android.ui.posts.PrepublishingViewModel
+import org.wordpress.android.viewmodel.Event
+
+class PrepublishingViewModelTest : BaseUnitTest() {
+    private lateinit var viewModel: PrepublishingViewModel
+
+    @Before
+    fun setup() {
+        viewModel = PrepublishingViewModel()
+    }
+
+    @Test
+    fun `when viewModel start is called with a null currentScreen, navigateToScreen should be invoked with HOME`() {
+        val expectedScreen = HOME
+
+        var event: Event<PrepublishingNavigationTarget>? = null
+        viewModel.navigationTarget.observeForever {
```

Comment:
@jd-alexander Just out of curiosity - is there any advantage in observing the target vs looking at the value in the assertion method?

```
        val expectedScreen = HOME
        viewModel.start(mock(), null)
        assertThat(viewModel.navigationTarget.value?.peekContent()?.targetScreen).isEqualTo(expectedScreen)
```

Reply:
Honestly, no 😅 I should do this. I am going to put it as a task in the code cleanup of the project. https://github.com/wordpress-mobile/WordPress-Android/issues/11751